[Update 8] - Add breaking changes details for u8 release note (#7586)
ballerina-platform/ballerina-dev-website: downloads/swan-lake-release-notes/swan-lake-2201.8.0/RELEASE_NOTE.md

## Backward-incompatible changes

> - A bug that resulted in not giving compilation errors for literal values due to invalidly inferred literal value types has been fixed.

Reviewer: ```suggestion
A bug related to the inference of numeric literal types has been fixed. 
```

> - A bug that resulted in not giving compilation errors for literal values, due to invalidly infered literal values types has been fixed.

Reviewer 2: ```suggestion
- A bug that resulted in not giving compilation errors for literal values due to invalidly inferred literal value types has been fixed.
```

    ```ballerina

>     2f|1 num = 2; // Compilation error now.

Reviewer: @SasinduDilshara @LakshanWeerasinghe, can we add a few more applicable scenarios here.

Reviewer 2: @LakshanWeerasinghe few examples are

```
2f|3d|6 num = 10; // Compilation error now.
3d|6 num = 10; // Compilation error now.
```

Author: Updated.

## Backward-incompatible changes

> - A bug related to the inference of numeric literal types has been fixed.

Reviewer 3: Please see https://github.com/ballerina-platform/ballerina-dev-website/pull/7642#discussion_r1305092375.

- Ballerina interoperability implementation may have an impact with the Java 17 support due to any incompatible changes. For example, Java 17 has some restrictions on using Java reflections with internal Java packages. For more information, see the Java 17 release notes.

- A bug related to deciding the types of numeric literals has been fixed.

    ```ballerina
    2f|1 a = 2; // Compilation error now, `2` is considered to be `int`.

>     3d|6 b = 3; // Compilation error now, `3` is considered to be `int`.

Reviewer 2: ```suggestion
    3d|6 b = 3; // Compilation error now, `3` is considered to be `int`.
    3d|4f|6 b = 4; // Compilation error now, `` is considered to be `int`.
```

- Ballerina interoperability implementation may have an impact with the Java 17 support due to any incompatible changes. For example, Java 17 has some restrictions on using Java reflections with internal Java packages. For more information, see the Java 17 release notes.

> - A bug related to the inference of numeric literal types has been fixed.

Reviewer 3: ```suggestion
- A bug related to deciding the types of numeric literals has been fixed.
```

    ```ballerina

>     2f|1 num1 = 2; // Compilation error now.
>     2f|3d|6 num2 = 10; // Compilation error now.
>     3d|6 num3 = 10; // Compilation error now.

Reviewer 3: ```suggestion
    2f|1 a = 2; // Compilation error now, `2` is considered to be `int`.
    3d|6 b = 3; // Compilation error now, `3` is considered to be `int`.
```

>     3d|6 num3 = 10; // Compilation error now.

Reviewer 3: These are already errors, not sure what the second one (L39) meant to show.

>     3d|4f|6 b = 4; // Compilation error now, `4` is considered to be `int`.

Reviewer 3: Similar to the first one? It's OK though.
